platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/PcesFileManagerTests.java

```diff
@@ -244,68 +257,64 @@ void incrementalPruningByGenerationTest() throws IOException {
+                    PcesFile.of(ancientMode, timestamp, sequenceNumber, lowerBound, upperBound, 0, fileDirectory);
 
-            minimumGeneration = random.nextLong(minimumGeneration, maximumGeneration + 1);
-            maximumGeneration =
-                    Math.max(maximumGeneration, random.nextLong(minimumGeneration, minimumGeneration + maxDelta));
-            timestamp = timestamp.plusMillis(random.nextInt(1, 100_000));
+            lowerBound = random.nextLong(lowerBound, upperBound + 1);
+            upperBound = Math.max(upperBound + 1, random.nextLong(lowerBound, lowerBound + maxDelta));
+            timestamp = timestamp.plusSeconds(random.nextInt(1, 100));
 
             files.add(file);
             createDummyFile(file);
         }
 
-        final PlatformContext platformContext = buildContext();
-
         // Choose a file in the middle. The goal is to incrementally purge all files before this file, but not
         // to purge this file or any of the ones after it.
         final int middleFileIndex = fileCount / 2;
 
         final PcesFile firstFile = files.getFirst();
         final PcesFile middleFile = files.get(middleFileIndex);
-        final PcesFile lastFile = files.getLast();
 
         // Set the clock before the first file is not garbage collection eligible
         final FakeTime time = new FakeTime(firstFile.getTimestamp().plus(Duration.ofMinutes(59)), Duration.ZERO);
 
-        final PcesFileTracker fileTracker =
-                PcesFileReader.readFilesFromDisk(buildContext(), TestRecycleBin.getInstance(), fileDirectory, 0, false);
-        final PcesFileManager manager = new PcesFileManager(platformContext, time, fileTracker, new NodeId(0), 0);
+        final PlatformContext platformContext = buildContext(ancientMode, time);
 
-        assertIteratorEquality(files.iterator(), fileTracker.getFileIterator(NO_MINIMUM_GENERATION, 0));
+        final PcesFileTracker fileTracker = PcesFileReader.readFilesFromDisk(
+                platformContext, TestRecycleBin.getInstance(), fileDirectory, 0, false, ancientMode);
+        final PcesFileManager manager = new PcesFileManager(platformContext, fileTracker, new NodeId(0), 0);
+
+        assertIteratorEquality(files.iterator(), fileTracker.getFileIterator(NO_LOWER_BOUND, 0));
 
         // Increase the timestamp a little at a time. We should gradually delete files up until
         // all files before the middle file have been deleted.
         final Instant endingTime =
                 middleFile.getTimestamp().plus(Duration.ofMinutes(60).minus(Duration.ofNanos(1)));
 
-        Duration nextTimeIncrease = Duration.ofSeconds(random.nextInt(1, 20));
-        while (time.now().plus(nextTimeIncrease).isBefore(endingTime)) {
-            time.tick(nextTimeIncrease);
-            manager.pruneOldFiles(lastFile.getMaximumGeneration() + 1);
-
+        while (time.now().isBefore(endingTime)) {
```

Comment:
Your changes here revert some recent changes I made to fix this unit test, which was flaky

Reply:
I re-did the merge for this test.